Drop Ubuntu 18.04; add Ubuntu 22.04 and 24.04

kenyon opened the pull request:
- metadata.json: allow puppetlabs-concat < 10
- metadata.json: drop EOL Ubuntu 18.04, add Ubuntu 20.04 and 22.04

kenyon committed:
metadata.json: drop EOL Ubuntu 18.04, add Ubuntu 22.04 and 24.04

TheMeier commented:
```
       	Jun 07 05:05:23 ubuntu2004-64-puppet7.example.com rsyslogd[4573]: error during parsing file /etc/rsyslog.d/50_rsyslog.conf, on or before line 3: syntax error on token '==' [v8.2001.0 try https://www.rsyslog.com/e/2207 ]
       	Jun 07 05:05:23 ubuntu2004-64-puppet7.example.com rsyslogd[4573]: could not interpret master config file '/etc/rsyslog.conf'. [v8.2001.0 try https://www.rsyslog.com/e/2207 ]
```
Maybe newer versions needs a different config?

kenyon commented:
@TheMeier thanks, I didn't notice that before. I'll try to look into it soon.

kenyon committed:
acceptance/actions: install rsyslog-elasticsearch on all OSes

elasticsearch modules have been split into this package on all
supported operating systems.
